### PR TITLE
@dzucconi -> add more params to partners, nest them in categories

### PR DESCRIPTION
**Pull request by 1aurabrown**
I'm putting the query I made here so that I don't forget it
```
{
  partner_categories(category_type:"Gallery"){
    id
    name
		primary: partners(eligible_for_primary_bucket: true, has_full_profile:true, default_profile_public:true) {
			... partner
    } 
    secondary: partners(eligible_for_secondary_bucket:true, has_full_profile:true, default_profile_public:true){
      ... partner
    }
  }
}

fragment partner on Partner  {
  id
  initials
  locations {
    city
  }
  profile {
    image {
      url(version:"featured")
    }
  }
}
```

Partners can take buckets and a bunch of other stuff. Also made it so that Partners nested in a Category can be filtered the same way.

### CODE REVIEW

File: schema/partners.js [hunk 1]
```diff
@@ -13,6 +14,20 @@ const Partners = {
       type: GraphQLString,
       description: 'Coordinates to find partners closest to',
     },
+    eligible_for_primary_bucket: {
+      type: GraphQLBoolean,
+      description: 'Indicates tier 1/2 for gallery, 1 for institution',
+    },
+    eligible_for_secondary_bucket: {
+      type: GraphQLBoolean,
+      description: 'Indicates tierl 3/4 for gallery, 2 for institution',
```
Comment (dzucconi) [inline schema/partners.js]: Typo: `tierl`

File: schema/partner_categories.js [hunk 2]
```diff
@@ -12,6 +13,9 @@ const PartnerCategories = {
     size: {
       type: GraphQLInt,
     },
+    category_type: {
+      type: GraphQLString,
+    },
```
Comment (dzucconi) [inline schema/partner_categories.js]: I think this should be an `EnumType` of possible values. Lemme know if you need halp setting that up

File: schema/partner_category.js [hunk 3]
```diff
@@ -24,12 +24,8 @@ const PartnerCategoryType = new GraphQLObjectType({
       type: GraphQLString,
     },
     partners: {
-      type: new GraphQLList(Partner.type),
-      args: {
-        size: {
-          type: GraphQLInt,
-        },
-      },
```
Comment (dzucconi) [inline schema/partner_category.js]: Just have to remove the two unused types from this file to pass lint

```
/home/runner/metaphysics/schema/partner_category.js
   9:3  error  "GraphQLList" is defined but never used  no-unused-vars
  10:3  error  "GraphQLInt" is defined but never used   no-unused-vars
```

I also highly recommend installing SublimeLinter, which will highlight these problems as you're editing

**Comment by dzucconi** [comment]
:balloon: :crystal_ball: :balloon:

File: schema/input_fields/partner_categorytype.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { GraphQLEnumType } from 'graphql';
+
+export default {
+  type: new GraphQLEnumType({
+    name: 'PartnerCategorytype',
```
Comment (1aurabrown) [inline schema/input_fields/partner_categorytype.js]: I lowercased this to indicate that it's not actually a `Type` but just something whose name includes the word "type". `PartnerCategoryType` is already a `Type` that represents a `PartnerCategory.` lmk if you think of a clearer way of naming this.

File: schema/partner_category.js (hunk 3 above)
Comment (1aurabrown) [inline schema/partner_category.js]: Installed it, but it doesn't seem to run when I modify or save a file... :(

Comment (dzucconi) [inline schema/partner_category.js]: Maybe restart Sublime? It should auto-run

**Pull request merged**